src/lib/y2storage/package_handler.rb

```diff
@@ -0,0 +1,168 @@
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "y2storage/used_storage_features"
+
+Yast.import "Package"
+Yast.import "PackagesProposal"
+Yast.import "Mode"
+Yast.import "Report"
+Yast.import "Popup"
+Yast.import "Label"
+
+module Y2Storage
+  #
+  # Class that takes care of packages in the storage context: Add
+  # storage-related packages to the set of packages to install during
+  # installation, install small sets of packages in the installed system (for
+  # the expert partitioner) as needed.
+  #
+  class PackageHandler
+    include Yast::Logger
+    include Yast::I18n
+
+    attr_reader :pkg_list
+
+    PROPOSAL_ID = "storage_proposal"
+
+    def initialize
+      textdomain("storage-ng")
+      reset
+    end
+
+    # Clear the package list
+    #
+    def reset
+      @pkg_list = []
+    end
+
+    # Add a number of packages to the list of packages to be installed
+    #
+    # @param  [Array<String>] package names
+    # @return [Array<String>] new package list (may contain duplicates)
+    #
+    def add_packages(pkg_list)
+      @pkg_list.concat(pkg_list)
+    end
+
+    # Add the packages for the storage features in 'devicegraph' to the list of
+    # packages to be installed
+    #
+    # @param [::Storage::devicegraph] devicegraph to obtain the features from
+    # @return [Array<String>] new package list (may contain duplicates)
+    #
+    def add_feature_packages(devicegraph)
+      used_features = UsedStorageFeatures.new(devicegraph)
+      add_packages(used_features.feature_packages)
+    end
+
+    # Start a package dependency resolver run
+    #
+    def solve
+      Yast::Pkg.PkgSolve(true)
+    end
+
+    # Commit the changes depending on the current mode: During OS installation,
+    # mark the package list for installation. In the installed system, install
+    # them immediately.
+    #
+    def commit
+      if Yast::Mode.installation
+        set_proposal_packages
+      else
+        install!
+      end
+    end
+
+    # Execute package installation. This will install the stored package list
+    # immediately, so it is not advisable to do this during the OS
+    # installation. In the latter case, use 'set_proposal_packages' instead.
+    #
+    # @return true on success, false on error
+    #
+    def install!
+      compact
+      return if @pkg_list.empty?
+      log.info("Installing #{pkg_list}")
+      success = Yast::Package.DoInstall(@pkg_list)
+      if !success
+        log.error("ERROR: Some packages could not be installed")
+        install_error_popup
```

Comment:
You are right, but this is taken literally from the old code. It also didn't check the return value. OTOH that means that this might also be a simpler error popup with just an "OK" button. There is not much the user can do at that point anyway.